Robot.java_03_14_17

```diff
@@ -1,418 +1 @@
 
-package org.usfirst.frc.team6418.robot;
-import java.util.*;
-import edu.wpi.first.wpilibj.IterativeRobot;
-import edu.wpi.first.wpilibj.command.Command;
-import edu.wpi.first.wpilibj.command.Scheduler;
-import edu.wpi.first.wpilibj.livewindow.LiveWindow;
-import edu.wpi.first.wpilibj.smartdashboard.SendableChooser;
-import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
-import edu.wpi.first.wpilibj.Joystick;
-import edu.wpi.first.wpilibj.RobotDrive;
-import edu.wpi.first.wpilibj.Spark;
-import edu.wpi.first.wpilibj.SpeedController;
-import edu.wpi.first.wpilibj.buttons.JoystickButton;
-import edu.wpi.cscore.CvSink;
-import edu.wpi.cscore.CvSource;
-import org.opencv.imgproc.Imgproc;
-import edu.wpi.cscore.UsbCamera;
-//import org.usfirst.frc.team6418.robot.commands.ExampleCommand;
-import edu.wpi.first.wpilibj.CameraServer;
-import edu.wpi.first.wpilibj.Encoder;
-import edu.wpi.first.wpilibj.Timer;
-import edu.wpi.first.wpilibj.ADXRS450_Gyro;
-
-import org.usfirst.frc.team6418.robot.subsystems.*;
-import org.opencv.core.*;
-import org.usfirst.frc.team6418.robot.commands.*;
-/**
- * The VM is configured to automatically run this class, and to call the
- * functions corresponding to each mode, as described in the IterativeRobot
- * documentation. If you change the name of this class or the package after
- * creating this project, you must also update the manifest file in the resource
- * directory.
- */
-public class Robot extends IterativeRobot {
-
-		
-	public static final int WINCHMOTOR = 2;
-	//public static OI oi;
-	public static RobotDrive driveTrain;
-
-	//public static RobotDrive driveTrain;
-	public static Winch winch;
-	public static GreenLightSystem greenlight = new GreenLightSystem();
-	Command autonomousCommand;
-	SendableChooser<Command> chooser = new SendableChooser<>();
-	//Testing UI, probably should put all joystick and buttons in a separate OI class
-	public static Joystick stick = new Joystick(0);
-	public static Joystick stick2 = new Joystick(1);
-	//for the second joystick
-	//need to find the right port
-	//changed button names-Anya
-	public static JoystickButton TriggerButton = new JoystickButton(stick,1); 
-	public static JoystickButton SideButton = new JoystickButton(stick,2);
-	//changed button to 3 to optimize for joystick (was 8)-Anya
-	public static JoystickButton HighButton = new JoystickButton(stick,5);
-	public static JoystickButton LowButton = new JoystickButton(stick, 3);
-	
-	//TEST - Anya
-	public static JoystickButton RightButton = new JoystickButton(stick,4);
-	
-	//use Trigger Button and Side Button , togggleWhenPressed(final Command command) 
-	//to toggle between arcadeDrive and tankDrive
-	//use TopButton to turn the green light on and off
-	
-	public static String driveMode;
-	public static String cameraMode;
```

Comment:
the static keyword in Java means that the field is shared by all instances (aka objects) of the class. So, as the code is, if you did:

a = new Robot()
b = new Robot()

then a.driveMode would be the exact same field as b.driveMode. Changing one, would change the other.

Normally, the only stuff that should be static are those things that belong to the class, not an object of the class. Note, if you have a static function, then the only stuff you can see are the static fields because a static function means that it doesn't have an object to work with.